### ISSUE TITLE
Listing nearby Bluetooth devices (mac) and WIFI networks (BSSID)

### ISSUE THREAD

**Issue by hllhll** [home-assistant/android#4210]
**Is your feature request related to a problem? Please describe.**
I would like to setup some automations such as 1) turning on my bluetooth when the phone detects my vehicle's Headunit ( assuming "wifi scanning always on" This is I think always running even if wifi is off). 2) Open the underground parking lot door w/o needing to use iBeacon, potentially based off wifi networks available nearby. (this is not a personal parking, rather a medium sized one, it takes a minute or so to reach the entrance)

**Describe the solution you'd like**
I would like to have ability (sensor) that would list WIFI network as seen in the wifi menu screen of my phone (even hidden ones if possible since I think their BSSID is broadcast).
The same goes for bluetooth, have to option to get the list of available bluetooth devices (macs, even w/o device names).

**Describe alternatives you've considered, if any**
So far, the app only allow for BLE beacons monitoring, and only "CONNECTED" bluetooth devices that are not beacons. Same goes for wifi, the wifi sensor will only get updated if I will connect to a specified network. 

The case in (1) above is actually more complex then that, the usecase is Wireless AA adapter; the Wifi on the adapter does not provide internet access, so the phone would not auto-connect (also mostly wifi is off for me at the time I enter the vehicle), hence, I cannot use the existing sensor. It will connect when the bluetooth first turns on and connect, then the phone would connect to the AA Adapter Wifi while getting the internet from the mobile network.

**Additional context**
As I understand, the information is available on the phone even if I'm not connected to the network. any reason why this is not already implemented? 

I have looked here and online for possible solutions, couldn't find any, if there's any other/3rd party ideas that can be integrated into ha/companion app will be glad to know.

**Comment by dshokouhi** [comment]
Part duplicate of #1536 please update this request to remove mentioning Bluetooth devices.

**Comment by macrojames** [comment]
I would like to add some usecases to this request (and also SSID in the title)

- informing my smart home that I am sitting in a specific plane/train/bus/tram type. These vehicles host a hotspot but are often limited in bandwidth, i don't connect to them. While SSID='XYZ' is in range, I am on that train. I can even calculate arrival time on train data. I can inform my family that I am on my way home.
- Moving inside a corporate building while not being connected to any access point to pin down my location in that building. I could set DND in a specific area or inform my colleagues that i am in the canteen.
- Any WiFi can be used to track your device location or zone without using GPS data or any other location data. You could even abuse some devices like esp8266 as beacons without BT/BLE

I could imagine many uses for the fields BSSID, SSID, type of security, maybe RSSI or channel.

**Comment by TechnoFairyGirl** [comment]
I'd like this in order to detect when I'm at work based on the existence of my work's wifi network. I don't actually connect to my work's wifi because it has a content filter so I'd rather just use mobile data. I'd be willing to implement it myself. I'll ask about it on the Discord. Or I could make a separate app to do it and push the results to MQTT.

**Comment by botts7** [comment]
This primitive has a **concrete downstream consumer** ready and shipped:

  - **HA Insights v1.21.x** ([WS handler
  source](https://github.com/botts7/ha-insights/blob/main/custom_components/ha_insights/ws_api/wifi_find_self.py)) accepts WiFi
  RSSI + BSSID samples via WebSocket and drives walking warmer/colder find-my-device UX in the **find-my-ha PWA**
  ([live](https://botts7.github.io/find-my-ha/)).
  - Today, only installs with controller integrations exposing per-client RSSI (UniFi, Mikrotik, some Asuswrt-Merlin firmwares)
  can use the WiFi find path. Asuswrt-stock + official `tplink_omada` + plain DHCP-based trackers — likely the majority of HA
  installs — are stuck. **The missing primitive is exactly what this issue proposes**: a sensor that lists nearby BSSIDs + their
  signal strength from the phone's perspective.

  **Use cases the same primitive unlocks** (beyond OP's parking / vehicle scenarios):

  - Walking find-my-device UX for non-BLE WiFi devices (the gap above)
  - Room-level presence inference via signal-to-AP gradient
  - Roaming diagnostics ("phone keeps re-associating with the wrong AP")
  - Hyper-local automations ("kitchen AP RSSI > -50 dBm → user is in kitchen", no PIR sensor needed)

  **Shape of a small reviewable PR** modeled on the existing BLE Beacon Monitor ([PR
  #2472](https://github.com/home-assistant/android/pull/2472) by @Adi146, migrated to common in [PR
  #3168](https://github.com/home-assistant/android/pull/3168)):

  - One sensor manager: `WifiNearbyApsSensorManager.kt`
  - State = nearest BSSID's RSSI (or count of visible APs)
  - Attributes = list of `{bssid, rssi, ssid, frequency, channel}` for all visible APs
  - Default cadence: standard sensor interval (current default behavior)
  - Optional foreground "active scan" mode: 1-2 Hz during a HA-Insights-initiated find session

  Architecture writeup with the full picture:
  [`docs/COMPANION_APP_PROPOSAL.md`](https://github.com/botts7/find-my-ha/blob/main/docs/COMPANION_APP_PROPOSAL.md). Happy to
  write the Kotlin if there's review appetite. @dshokouhi @Adi146 — does the Beacon Monitor's structure translate cleanly as a
  template here?

**Comment by TimoPtr** [comment]
I'm not sure to follow your proposal for the app since it mix multiples pieces, I don't think it is possible to have Wifi scan at 1-2Hz at all.

**Comment by botts7** [comment]
Fair point on both — the 1-2 Hz figure was wrong (Android's scan throttling caps it well below that) and the comment tried to 
  cover too much.
 
  You'd know far better than me what's actually viable inside the Companion app. Is there a smaller shape — even just passive 
  listening to whatever scans the system itself triggers — that you'd consider worth exploring? Happy to follow your lead on 
  what's realistic, or drop the proposal entirely if the platform constraints make it impractical
